sdk/azcore/tracing/tracing.go

```diff
@@ -0,0 +1,168 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// Package tracing contains the definitions needed to support distributed tracing.
+package tracing
+
+import (
+	"context"
+)
+
+// ProviderOptions contains the optional values when creating a Provider.
+type ProviderOptions struct {
+	// for future expansion
+}
+
+// NewProvider creates a new Provider with the specified values.
+//   - impl is the underlying implementation for creating Tracer instances
+//   - options contains optional values; pass nil to accept the default value
+func NewProvider(impl func(name, version string) Tracer, options *ProviderOptions) Provider {
```

Comment:
```suggestion
func NewProvider(newTracerFn func(name, version string) Tracer, options *ProviderOptions) Provider {
```
It'd be nice if it was styled a bit like a "constructor" function.

Reply:
Good idea, fixed.